hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -362,6 +362,9 @@ private void setHmsTableParameters(String newMetadataLocation, Table tbl, Map<St
     // remove any props from HMS that are no longer present in Iceberg table props
     obsoleteProps.forEach(parameters::remove);
 
+    // remove the DDL_TIME so it gets refreshed
```

Comment:
Nit: you might consider formatting this comment more generically like the one above. Something like `// remove any props that are controlled by HMS so it can handle them`. That way it's also clear where to add more properties like that in the future.

Up to you though. There's nothing wrong per-se with the current comment. Just this has come up a few times on Slack and I'm not sure if there's other properties that might fall into this category.